library/options.inc.php

```diff
@@ -2861,6 +2883,14 @@ function generate_display_field($frow, $currvalue)
             }
             $i++;
         }
+    } elseif ($data_type == 53) {
+        $service = new EncounterService();
+        if (!empty($currvalue)) {
+            $encounter = $service->getEncounterById($currvalue);
+            if (!empty($encounter)) {
+                $s = $row['date'] ?? '';
```

Comment:
can't figure out where `$row['date']` is coming from

Reply:
also whatever $s ends up being should be html escaped

Reply:
Yup, that was a bug, needed to be $encounter.  I assumed $s was escaped where it was consumed since $data_type 52 was skipping the escaping.  I looked at the other data types and it looks like everything is escaped in this function so I'll fix the missing text() on $data_type 52 as well.